src/filters/qualitative-filter.js

```diff
@@ -0,0 +1,221 @@
+    e.stopPropagation()
+
+    let arr = (optionsValue || '').split(',').filter((v) => v)
+    if (!arr.length) {
+      arr = [...options]
+    }
+
+    const i = arr.indexOf(opt)
+    if (i > -1) {
+      arr.splice(i, 1)
+    } else {
+      arr.push(opt)
+    }
+
+    if (!optionsValue || optionsValue.includes(opt)) {
+      onlyButtonContainer[options.indexOf(opt)].style.display = 'none'
+    } else {
+      onlyButtonContainer[options.indexOf(opt)].style.display = 'initial'
+    }
+
+    setOptionsValue(arr.length && arr.length < options.length ? arr.join(',') : undefined)
+  }
+
+  const handleOnlyOnClick = (e, opt) => {
+    e.stopPropagation()
+    setOptionsValue(opt)
+  }
+
+  const applyOnClick = (e) => {
+    e.stopPropagation()
+    closePopper && closePopper(e)
+    setFilter(optionsValue)
+  }
+
+  const cancelOnClick = (e) => {
+    e.stopPropagation()
+    closePopper && closePopper(e)
+    setOptionsValue(filterValue)
+  }
+
+  return (
+    <div>
+      <List ref={listItemRef} className={classes.list}>
+        <TextField fullWidth variant='outlined'
+          size='small'
+          InputProps={{
+            endAdornment: (
+              <InputAdornment position='end'>
+                <SearchIcon />
+              </InputAdornment>
+            ),
+            'aria-label': 'search',
+          }}
+          onClick={(e) => { e.stopPropagation() }}
+          onChange={handleOnSearch}
+          onKeyUp={filterList}
+          value={value || ''}
+          placeholder={`Search in ${options.length} records...`}
+        />
+        {options.map((opt, index) => {
+          const labelID = `${id}-selection-label-${opt}`
+          return (
+            <div key={opt} className={classes.listItemContainer} onMouseEnter={() => handleOnMouseEnter(opt, index)} onMouseLeave={() => handleOnMouseLeave(index)}>
+              <ListItem
+                className={classes.MuiListItem}
+                role={undefined}
+                dense
+                button
+                onClick={(e) => {
+                  handleListItemOnClick(e, opt)
+                }}
+              >
+                <ListItemIcon className={classes.MuiListItemIcon}>
+                  <Checkbox
+                    color='primary'
+                    edge='start'
+                    checked={!optionsValue || (optionsValue || '').includes(opt)}
+                    tabIndex={-1}
+                    disableRipple
+                    inputProps={{ 'aria-labelledby': labelID }}
+                  />
+                </ListItemIcon>
+                <ListItemText className='listItemText' id={labelID} primary={opt} />
+              </ListItem>
+
+              <Button
+                className={classes.onlyButton}
+                color="primary"
+                onClick={(e) => {
+                  handleOnlyOnClick(e, opt)
+                }}
+              >
+                Only
+              </Button>
```

Comment:
we can nest this one inside `ListItem` for better organization and consistent styling. Alternatively we can take out the hover animation from `ListItem`